### PR TITLE
fix: filter out new in java16+ method(s) from proxy class for by-builder

### PR DESCRIPTION
**Pull request by ayuryev**
… AppiumByBuilder. [#1575](https://github.com/appium/java-client/issues/1575)

## Change list
Filter out proxyClassLookup method from Proxy class (for Java 16+) in AppiumByBuilder. [#1575](https://github.com/appium/java-client/issues/1575)
 
## Types of changes
- [ ] No changes in production code.
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
Avoid getting java.lang.NoSuchMethodException when loading PageFactory elements in Java16+

### CODE REVIEW

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java [hunk 1]
```diff
@@ -73,13 +78,13 @@ protected AppiumByBuilder(String platform, String automation) {
...
 
     private static Method[] prepareAnnotationMethods(Class<? extends Annotation> annotation) {
-        List<String> targeAnnotationMethodNamesList =
+        List<String> targetAnnotationMethodNamesList =
             getMethodNames(annotation.getDeclaredMethods());
-        targeAnnotationMethodNamesList.removeAll(METHODS_TO_BE_EXCLUDED_WHEN_ANNOTATION_IS_READ);
-        Method[] result = new Method[targeAnnotationMethodNamesList.size()];
-        for (String methodName : targeAnnotationMethodNamesList) {
+        targetAnnotationMethodNamesList.removeAll(METHODS_TO_BE_EXCLUDED_WHEN_ANNOTATION_IS_READ);
+        Method[] result = new Method[targetAnnotationMethodNamesList.size()];
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java:84]: could this be done in more functional style (e.g. using streams)?

Comment (ayuryev) [reply src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java:82]: Sure, done.

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java [hunk 2]
```diff
@@ -45,13 +48,13 @@
...
-                    getMethodNames(Annotation.class.getDeclaredMethods());
-                annotationClassMethodNames.removeAll(objectClassMethodNames);
-                addAll(annotationClassMethodNames);
+                List<String> excludeMethodNames =
+                    Stream.of(getMethodNames(Annotation.class.getDeclaredMethods()),
+                            getMethodNames(Proxy.class.getDeclaredMethods()))
+                        .flatMap(List::stream)
+                    .filter(m -> !getMethodNames(Object.class.getDeclaredMethods()).contains(m))
+                    .collect(Collectors.toList());
+                addAll(excludeMethodNames);
```
Comment (valfirst) [inline src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java:57]:
```suggestion
                List<String> objectClassMethodNames = getMethodNames(Object.class.getDeclaredMethods());
                Stream.of(Annotation.class, Proxy.class)
                    .map(Class::getDeclaredMethods)
                    .map(AppiumByBuilder::getMethodNames)
                    .flatMap(List::stream)
                    .filter(m -> !objectMethodNames.contains(m))
                    .forEach(this::add);
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -102,6 +102,8 @@ dependencies {
   - Remove setElementValue helper. [#1577](https://github.com/appium/java-client/pull/1577)
   - Remove selenium package override. [#1555](https://github.com/appium/java-client/pull/1555)
   - remove redundant exclusion of Gradle task signMavenJavaPublication. [#1568](https://github.com/appium/java-client/pull/1568)
+- **[BUG FIX]
+  - Filter out proxyClassLookup method from Proxy class (for Java 16+) in AppiumByBuilder. [#1575](https://github.com/appium/java-client/issues/1575)
```
Comment (valfirst) [inline README.md:106]: I don't think it should be in `8.0.0-beta` since it's released

**Review by valfirst — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (ayuryev) [reply README.md:106]: 8.0.1-beta? Should I bump the version?

File: README.md [hunk 4]
```diff
@@ -66,6 +66,10 @@ dependencies {
 ```
 
 ## Changelog
+*8.0.1-beta*
```
Comment (mykola-mokhnach) [inline README.md:69]: Please revert the changes to this file and to the build.gradle. 
These files are updated automatically by the publishing script.

Comment (ayuryev) [reply README.md]: Done

**Comment by margasan** [comment]
@mykola-mokhnach could you merge this fix into 7.6.x version?
